torch/csrc/jit/frontend/schema_emitter.cpp

```diff
@@ -0,0 +1,144 @@
+#include <torch/csrc/jit/frontend/schema_emitter.h>
+
+#include <torch/csrc/jit/frontend/builtin_functions.h>
+#include <torch/csrc/jit/frontend/error_report.h>
+
+#include <torch/csrc/jit/ir/named_value.h>
+
+namespace torch {
+namespace jit {
+
+// pack outputs of a function following python rules. If there is a single value
+// return a SimpleValue, otherwise pack all the values into a Tuple.
+static Value* packOutputs(
+    Graph& g,
+    at::ArrayRef<Value*> values,
+    c10::OptNameList field_names) {
+  if (values.size() == 1) {
+    return values[0];
+  }
+  std::shared_ptr<FunctionSchema> schema;
+  TupleTypePtr named_tuple = nullptr;
+  if (field_names) {
+    auto types = fmap(values, [](Value* v) { return v->type(); });
+    named_tuple =
+        TupleType::createNamed(c10::nullopt, field_names.value(), types);
+  }
+  return g.insertNode(g.createTuple(values, named_tuple))->output();
+}
+
```

Comment:
Are `packOutputs`, `emitBuiltinNode` and `emitBuiltinCall` just moved or moved and modified? I didn't spot any differences, assuming they are not modified, thus not carefuly reviewed. Let me know if this is not the case.

Reply:
Just moved :)